### PR TITLE
TDL 22684/sync all form ids and handle no questions error

### PR DESCRIPTION
**Pull request by kethan1122**
# Description of change
Handles/Fixes:
- Syncs data for all forms if no form ID is provided in config.
- Currently, tap fails with an following error when there are no questions associated with a form. Handles this scenario by logging a statement and continues to sync data for other streams.
- Updates unittests
```
CRITICAL 'fields'
Traceback (most recent call last):
  File "/Users/Documents/project/taps/tap-typeform/tap_typeform/__init__.py", line 54, in <module>
    main()
  File "/Users/Documents/project/taps/tap-typeform/typeform-venv/lib/python3.9/site-packages/singer_python-5.10.0-py3.9.egg/singer/utils.py", line 229, in wrapped
    return fnc(*args, **kwargs)
  File "/Users/Documents/project/taps/tap-typeform/tap_typeform/__init__.py", line 51, in main
    _sync(client, config, args.state, catalog.to_dict(), valid_forms)
  File "/Users/Documents/project/taps/tap-typeform/typeform-venv/lib/python3.9/site-packages/tap_typeform-2.0.1-py3.9.egg/tap_typeform/sync.py", line 76, in sync
    stream_obj.sync_obj(client, state, catalog['streams'], form, config["start_date"],
  File "/Users/Documents/project/taps/tap-typeform/typeform-venv/lib/python3.9/site-packages/tap_typeform-2.0.1-py3.9.egg/tap_typeform/streams.py", line 176, in sync_obj
    for record in response[self.data_key]:
KeyError: 'fields'
```

# Manual QA steps
 - Tested without having `forms` field in config and noticed data being extracted for all the forms.
 - Tested with `forms` field value as `""` and `None`  and noticed data being extracted for all the forms.
 - Tested with single valid `form_id` in config and noticed data being extracted only for that single form id.
 - Tested with multiple valid `form_ids` which are comma separated in config and noticed data being extracted only for those form IDs
 - Created a form with no questions and made sure tap logs a statement which reads `There are no questions associated with form` without failing.
 
# Risks
 - Low, all the existing connections should be having `forms` field configured. This changes will not have any impact on them
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_typeform/__init__.py [hunk 1]
```diff
@@ -20,33 +20,35 @@ def validate_form_ids(client, config):
     """Validate the form ids passed in the config"""
     form_stream = Forms()
 
+    api_forms = {form.get('id') for res in form_stream.get_forms(client) for form in res if form}
```
Comment (dsprayberry) [inline tap_typeform/__init__.py:23]: This should probably be within the "if not" block on 24 to make sure we're not getting all forms even when a user has provided forms.

File: tests/unittests/test_main.py [hunk 2]
```diff
@@ -101,21 +105,22 @@ def test_no_form_given(self, mock_forms):
         Test when no forms are given in config, an error is raised with an expected message.
         """
         config = {}
-        with self.assertRaises(NoFormsProvidedError) as e:
+        with self.assertLogs(level='INFO') as log_statement:
```
Comment (dsprayberry) [inline tests/unittests/test_main.py:108]: we should also add assertions around ensuring that api_forms is called / returned as expected, even if we mock the responses.

**Review by dsprayberry — CHANGES_REQUESTED** [review]
We can remove the NoFormsProvidedError class since it will no longer be used.

We should probably move api_forms into the if not config.get('forms') to ensure we're not making extraneous calls when we already have user provided forms.

Can we also add unit test assertions around fetching forms  (via api_forms) when none are provided in addition to raising the log warning?

File: tap_typeform/__init__.py [hunk 3]
```diff
@@ -20,33 +20,35 @@ def validate_form_ids(client, config):
...
     if not config.get('forms'):
-        raise NoFormsProvidedError("No forms were provided in the config")
```
[the remark this answers is not in the document]
Comment (kethan1122) [reply tap_typeform/__init__.py:24]: Removed

File: tap_typeform/__init__.py (hunk 1 above)
Comment (kethan1122) [reply tap_typeform/__init__.py:21]: This is also needed to validate the form ids configured by the user, so keeping it as it is

File: tests/unittests/test_main.py (hunk 2 above)
Comment (kethan1122) [reply tests/unittests/test_main.py:113]: added

File: tap_typeform/__init__.py (hunk 1 above)
Comment (aaronbannin) [reply tap_typeform/__init__.py:21]: Agree with @kethan1122 ; this step is used to verify that the form id's provided are valid ids. It now is also used to capture all forms from the account.

**Comment by aaronbannin** [comment]
Thanks @kethan1122 and @dsprayberry!
